### ISSUE TITLE
cg clean crash for balsamic and mutant

### ISSUE THREAD

**Issue by pbiology** [Clinical-Genomics/cg#1596]
## Description

cg clean crashes both for `cg clean mutant-past-run-dirs` and `cg clean balsamic-past-run-dirs`. This becomes quite apparent with the new systemd services for both of these.

The errors seem to be quite similar (same?) for both

### Balsamic
```
θ68° 1d [hiseq.clinical@hasta:/home/proj/stage/servers/config/hasta.scilifelab.se/systemd] [base] migrate_cg_clean_systemd(13)+ 3 ± systemctl --user status -l -n 15 cg-clean-analysis-balsamic.service
● cg-clean-analysis-balsamic.service - Removal of old balsamic analysis files
   Loaded: loaded (/home/hiseq.clinical/.config/systemd/user/cg-clean-analysis-balsamic.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2022-10-10 12:04:23 CEST; 35min ago
  Process: 238517 ExecStart=/bin/bash -c /home/proj/bin/conda/envs/P_main/bin/cg      --config /home/proj/production/servers/config/hasta.scilifelab.se/cg.yaml      clean      balsamic-past-run-dirs      `date -d -45days +%%Y-%%m-%%d`      -y (code=exited, status=1/FAILURE)
 Main PID: 238517 (code=exited, status=1/FAILURE)

Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.meta.workflow.analysis[238517] INFO Case lightbull exists in status db
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.meta.workflow.analysis[238517] ERROR Working directory path for lightbull does not exist
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.cli.workflow.commands[238517] ERROR Failed to clean directories for case lightbull -
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.cli.workflow.commands[238517] INFO Cleaning balsamic output for primesnake
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.meta.workflow.analysis[238517] INFO Case primesnake exists in status db
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.meta.workflow.analysis[238517] ERROR Working directory path for primesnake does not exist
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.cli.workflow.commands[238517] ERROR Failed to clean directories for case primesnake -
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.cli.workflow.commands[238517] INFO Cleaning balsamic output for maturecat
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.meta.workflow.analysis[238517] INFO Case maturecat exists in status db
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.meta.workflow.analysis[238517] ERROR Working directory path for maturecat does not exist
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.cli.workflow.commands[238517] ERROR Failed to clean directories for case maturecat -
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.cli.workflow.commands[238517] INFO Cleaning balsamic output for stablehyena
Oct 10 12:04:16 hasta.scilifelab.se bash[238517]: 2022-10-10 12:04:16 hasta.scilifelab.se cg.meta.workflow.analysis[238517] INFO Case stablehyena exists in status db
Oct 10 12:04:23 hasta.scilifelab.se systemd[5702]: Failed to start Removal of old balsamic analysis files.
Oct 10 12:04:23 hasta.scilifelab.se systemd[5702]: cg-clean-analysis-balsamic.service failed.
```

### Mutant
```
θ70° 1d [hiseq.clinical@hasta:~] [base] 3 $ systemctl --user status -l -n15 cg-clean-analysis-mutant.service
● cg-clean-analysis-mutant.service - Removal of old mutant analysis files
   Loaded: loaded (/home/hiseq.clinical/.config/systemd/user/cg-clean-analysis-mutant.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2022-10-10 12:27:51 CEST; 13min ago
  Process: 19979 ExecStart=/bin/bash -c /home/proj/bin/conda/envs/P_main/bin/cg      --config /home/proj/production/servers/config/hasta.scilifelab.se/cg.yaml      clean      mutant-past-run-dirs      `date -d -21days +%%Y-%%m-%%d`      -y (code=exited, status=1/FAILURE)
 Main PID: 19979 (code=exited, status=1/FAILURE)

Oct 10 12:27:50 hasta.scilifelab.se bash[19979]: 2022-10-10 12:27:50 hasta.scilifelab.se cg.cli.workflow.commands[19979] ERROR Failed to clean directories for case restedraptor -
Oct 10 12:27:50 hasta.scilifelab.se bash[19979]: 2022-10-10 12:27:50 hasta.scilifelab.se cg.cli.workflow.commands[19979] INFO Cleaning sars-cov-2 output for quickwolf
Oct 10 12:27:50 hasta.scilifelab.se bash[19979]: 2022-10-10 12:27:50 hasta.scilifelab.se cg.meta.workflow.analysis[19979] INFO Case quickwolf exists in status db
Oct 10 12:27:50 hasta.scilifelab.se bash[19979]: 2022-10-10 12:27:50 hasta.scilifelab.se cg.meta.workflow.analysis[19979] ERROR Working directory path for quickwolf does not exist
Oct 10 12:27:50 hasta.scilifelab.se bash[19979]: 2022-10-10 12:27:50 hasta.scilifelab.se cg.cli.workflow.commands[19979] ERROR Failed to clean directories for case quickwolf -
Oct 10 12:27:50 hasta.scilifelab.se bash[19979]: 2022-10-10 12:27:50 hasta.scilifelab.se cg.cli.workflow.commands[19979] INFO Cleaning sars-cov-2 output for maturegoose
Oct 10 12:27:50 hasta.scilifelab.se bash[19979]: 2022-10-10 12:27:50 hasta.scilifelab.se cg.meta.workflow.analysis[19979] INFO Case maturegoose exists in status db
Oct 10 12:27:50 hasta.scilifelab.se bash[19979]: 2022-10-10 12:27:50 hasta.scilifelab.se cg.meta.workflow.analysis[19979] ERROR Working directory path for maturegoose does not exist
Oct 10 12:27:50 hasta.scilifelab.se bash[19979]: 2022-10-10 12:27:50 hasta.scilifelab.se cg.cli.workflow.commands[19979] ERROR Failed to clean directories for case maturegoose -
Oct 10 12:27:50 hasta.scilifelab.se bash[19979]: Aborted!
Oct 10 12:27:51 hasta.scilifelab.se systemd[5702]: cg-clean-analysis-mutant.service: main process exited, code=exited, status=1/FAILURE
Oct 10 12:27:51 hasta.scilifelab.se systemd[5702]: Failed to start Removal of old mutant analysis files.
Oct 10 12:27:51 hasta.scilifelab.se systemd[5702]: Unit cg-clean-analysis-mutant.service entered failed state.
Oct 10 12:27:51 hasta.scilifelab.se systemd[5702]: Triggering OnFailure= dependencies of cg-clean-analysis-mutant.service.
Oct 10 12:27:51 hasta.scilifelab.se systemd[5702]: cg-clean-analysis-mutant.service failed.
```

The behaviour also appears when just running cg on its own.

## Suggested solution

Should probably look into cg to undrstand what is happening here. I'm afraid I don't have the knowledge of cg to quite understand where this is coming from.

## This can be closed when

- [ ] systemd services for `cg clean mutant-past-run-dirs` and `cg clean balsamic-past-run-dirs` can be executed without fail

## Blocked by

*If there are any blocking issues/prs/things in this or other repos. Please link to them.*

**Comment by henrikstranneheim** [comment]
@pbiology Could it be permission errors? For instance, someone has started the analysis as their own user and not hiseq.clinical? Or related to: https://github.com/Clinical-Genomics/cg/issues/1574. I.e. that the case path does not extiist anymore and this is not in sync with the database

**Comment by pbiology** [comment]
Tried to poke around and couldn't see any files to some example cases in `/home/proj/stage/housekeeper-bundles`

So it seems possible that the #1574 is the cause here like you said.
Will add that issues as blocker and then re-try this once that is merged

**Comment by pbiology** [comment]
Also the cleaning for old mip runs is affected by this. Just adding here to remember

**Comment by pbiology** [comment]
Might be solved by https://github.com/Clinical-Genomics/cg/pull/1577

**Comment by henrikstranneheim** [comment]
### MIP

```

cg-clean-analysis-mip.service - Removal of files related to the analyses
  Loaded: loaded (/home/hiseq.clinical/.config/systemd/user/cg-clean-analysis-mip.service; enabled; vendor preset: enabled)
  Active: failed (Result: exit-code) since Fri 2022-10-21 01:40:00 CEST; 33ms ago
 Process: 26376 ExecStart=/bin/bash -c /home/proj/bin/conda/envs/P_main/bin/cg      --config /home/proj/production/servers/config/[hasta.scilifelab.se/cg.yaml](http://hasta.scilifelab.se/cg.yaml)      clean      mip-past-run-dirs      `date -d -21days +%%Y-%%m-%%d`      -y (code=exited, status=1/FAILURE)
Main PID: 26376 (code=exited, status=1/FAILURE)

Oct 21 01:34:06 [hasta.scilifelab.se](http://hasta.scilifelab.se/) systemd[5702]: Starting Removal of files related to the analyses...
Oct 21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) [cg.apps.housekeeper.hk](http://cg.apps.housekeeper.hk/)[26376] WARNING Called undefined __fields__ on HousekeeperAPI, please wrap
Oct 21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) [cg.apps.housekeeper.hk](http://cg.apps.housekeeper.hk/)[26376] WARNING Called undefined __fields__ on HousekeeperAPI, please wrap
Oct 21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) [cg.apps.housekeeper.hk](http://cg.apps.housekeeper.hk/)[26376] WARNING Called undefined __fields__ on HousekeeperAPI, please wrap
Oct 21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) [cg.apps.housekeeper.hk](http://cg.apps.housekeeper.hk/)[26376] WARNING Called undefined __fields__ on HousekeeperAPI, please wrap
Oct 21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) [cg.apps.housekeeper.hk](http://cg.apps.housekeeper.hk/)[26376] WARNING Called undefined __fields__ on HousekeeperAPI, please wrap
Oct 21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) [cg.apps.housekeeper.hk](http://cg.apps.housekeeper.hk/)[26376] WARNING Called undefined __fields__ on HousekeeperAPI, please wrap
Oct 21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) [cg.apps.housekeeper.hk](http://cg.apps.housekeeper.hk/)[26376] WARNING Called undefined __fields__ on HousekeeperAPI, please wrap
Oct 21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:34:27 [hasta.scilifelab.se](http://hasta.scilifelab.se/) [cg.apps.housekeeper.hk](http://cg.apps.housekeeper.hk/)[26376] WARNING Called undefined __fields__ on HousekeeperAPI, please wrap
Oct 21 01:38:52 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:38:52 [hasta.scilifelab.se](http://hasta.scilifelab.se/) cg.cli.workflow.commands[26376] INFO Cleaning 342 analyses created before 2022-09-30 00:00:00
Oct 21 01:38:55 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:38:55 [hasta.scilifelab.se](http://hasta.scilifelab.se/) cg.cli.workflow.commands[26376] INFO Cleaning mip-dna output for noblebear
Oct 21 01:38:55 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:38:55 [hasta.scilifelab.se](http://hasta.scilifelab.se/) cg.meta.workflow.analysis[26376] INFO Case noblebear exists in status db
Oct 21 01:38:58 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:38:58 [hasta.scilifelab.se](http://hasta.scilifelab.se/) cg.meta.workflow.analysis[26376] INFO Cleaned /home/proj/production/rare-disease/cases/noblebear
Oct 21 01:38:58 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:38:58 [hasta.scilifelab.se](http://hasta.scilifelab.se/) cg.meta.workflow.analysis[26376] INFO Adding a cleaned at date for case noblebear
Oct 21 01:38:58 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:3
.
.
.
Oct 21 01:40:00 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:40:00 [hasta.scilifelab.se](http://hasta.scilifelab.se/) cg.meta.workflow.analysis[26376] INFO Cleaned /home/proj/production/rare-disease/cases/cleverbull
Oct 21 01:40:00 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: 2022-10-21 01:40:00 [hasta.scilifelab.se](http://hasta.scilifelab.se/) cg.meta.workflow.analysis[26376] INFO Adding a cleaned at date for case cleverbull
Oct 21 01:40:00 [hasta.scilifelab.se](http://hasta.scilifelab.se/) bash[26376]: Aborted!
Oct 21 01:40:00 [hasta.scilifelab.se](http://hasta.scilifelab.se/) systemd[5702]: cg-clean-analysis-mip.service: main process exited, code=exited, status=1/FAILURE
Oct 21 01:40:00 [hasta.scilifelab.se](http://hasta.scilifelab.se/) systemd[5702]: Failed to start Removal of files related to the analyses.
Oct 21 01:40:00 [hasta.scilifelab.se](http://hasta.scilifelab.se/) systemd[5702]: Unit cg-clean-analysis-mip.service entered failed state.
Oct 21 01:40:00 [hasta.scilifelab.se](http://hasta.scilifelab.se/) systemd[5702]: Triggering OnFailure= dependencies of cg-clean-analysis-mip.service.
Oct 21 01:40:00 [hasta.scilifelab.se](http://hasta.scilifelab.se/) systemd[5702]: cg-clean-analysis-mip.service failed.
```

II think we have partially fixed the issue. Now analyses path that are non existing are handled correctly and get a cleaned_date. However, now we have an issue with cases without samples:

```
Oct 21 10:25:54 hasta.scilifelab.se bash[263235]: 2022-10-21 10:25:54 hasta.scilifelab.se cg.meta.workflow.analysis[263235] ERROR Case amplebee has no samples in in StatusDB!
Oct 21 10:25:54 hasta.scilifelab.se bash[263235]: 2022-10-21 10:25:54 hasta.scilifelab.se cg.cli.workflow.commands[263235] ERROR Failed to clean directories for case amplebee -
Oct 21 10:25:54 hasta.scilifelab.se bash[263235]: Aborted!
Oct 21 10:25:55 hasta.scilifelab.se systemd[5702]: cg-clean-analysis-balsamic.service: main process exited, code=exited, status=1/FAILURE
Oct 21 10:25:55 hasta.scilifelab.se systemd[5702]: Failed to start Removal of old balsamic analysis files.
Oct 21 10:25:55 hasta.scilifelab.se systemd[5702]: Unit cg-clean-analysis-balsamic.service entered failed state.
Oct 21 10:25:55 hasta.scilifelab.se systemd[5702]: Triggering OnFailure= dependencies of cg-clean-analysis-balsamic.service.
Oct 21 10:25:55 hasta.scilifelab.se systemd[5702]: cg-clean-analysis-balsamic.service failed.
``` 
<img width="1337" alt="Screenshot 2022-10-21 at 10 36 30" src="https://user-images.githubusercontent.com/584738/197151695-53459bef-7369-437b-a2be-e5e408d8f145.png">

<img width="1337" alt="Screenshot 2022-10-21 at 10 37 42" src="https://user-images.githubusercontent.com/584738/197152005-ebc82e98-cf86-42aa-98a7-24c77a945642.png">

@moahaegglund @karlnyr I cannot think of any reason to have cases without samples - do you?

**Comment by pbiology** [comment]
me and @GustavLagneborgScilifelab was also looking at this today and at least in terms of automation we're now only getting "correct" errors. So in terms of services these seem to work correctly now

**Comment by henrikstranneheim** [comment]
@GustavLagneborgScilifelab  @pbiology I think we close this one and open a new issue describing the case without sample error
